eth/stagedsync/stage_bor_heimdall.go

```diff
@@ -593,9 +593,6 @@ func persistValidatorSets(
 			snap = s
 			break
 		}
-		if chain != nil && blockNum < chain.FrozenBlocks() {
```

Comment:
just FYI: in all such comparisons `blockNum < chain.FrozenBlocks()` will need add `chain.FrozenBlocks() > 0` check.